test/e2e/kubernetes/pod/pod.go

```diff
@@ -629,7 +605,9 @@ func WaitOnSucceeded(name, namespace string, sleep, timeout time.Duration) (bool
 			pod = result.pod
 			mostRecentWaitOnSucceededPodError = result.err
 			if mostRecentWaitOnSucceededPodError == nil {
-				return pod.IsSucceeded(), nil
+				if pod.IsSucceeded() {
```

Comment:
This actually fixes a bug :/

We don't want to return a `false, nil` result in such a func. A non-truthy succeeded result should always accompany an error.